identity-server/src/IdentityServer/Endpoints/DiscoveryEndpoint.cs

```diff
@@ -63,8 +65,31 @@ public async Task<IEndpointResult> ProcessAsync(HttpContext context)
 
         // generate response
         _logger.LogTrace("Calling into discovery response generator: {type}", _responseGenerator.GetType().FullName);
-        var response = await _responseGenerator.CreateDiscoveryDocumentAsync(baseUrl, issuerUri);
+    
+#pragma warning disable DUENDEPREVIEW001
+        if (_options.Preview.EnableDiscoveryDocumentCache)
+#pragma warning restore DUENDEPREVIEW001
+        {
+            // we need a cache but since this is preview, we don't
+            // want to force folks to register IMemoryCache if they aren't caching.
+            // This requires we use the service locator pattern :(
+            var cache = context.RequestServices.GetService<IMemoryCache>();
```

Comment:
The IS code already has `IDistributedCache` registered by default I believe. Can you see if that's true and perhaps switch to using that?

Reply:
There are warnings around `IDisributedCache` in IdentityServer that hint that folks shouldn't use the `MemoryDistributedCache`, so I can't assume that an instance of `IDisributedCache` will be in memory.

https://github.com/DuendeSoftware/products/blob/511f1d5a7d54cc121464a0d79c57634c972722d3/identity-server/src/IdentityServer/Configuration/DependencyInjection/ConfigureOpenIdConnectOptions.cs#L39-L47

Reply:
This is a good point, and I like what you did checking that IMemoryCache resolves. Let's keep with IMemoryCache.